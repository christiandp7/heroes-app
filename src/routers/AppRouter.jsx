import React, { useContext } from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import { AuthContext } from '../auth/AuthContext'
// import Navbar from "../components/ui/Navbar";
import LoginScreen from '../components/login/LoginScreen'
// import MarvelScreen from "../components/marvel/MarvelScreen";
import DashboardRoutes from './DashboardRoutes'
import PrivateRoute from './PrivateRoute'
import PublicRoute from './PublicRoute'

const AppRouter = () => {
	const { user } = useContext(AuthContext)
	return (
		<Router>
			<div>
				<Switch>
					<PublicRoute
						exact
						path="/login"
						component={LoginScreen}
						isAuthenticated={user.logged}
					/>

					<PrivateRoute
						path="/"
						component={DashboardRoutes}
						isAuthenticated={user.logged}
					/>
				</Switch>
			</div>
		</Router>
	)
}

export default AppRouter
