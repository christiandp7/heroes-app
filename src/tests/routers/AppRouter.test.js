import { mount } from 'enzyme/build'
import AppRouter from '../../routers/AppRouter'
import { AuthContext } from '../../auth/AuthContext'
// import { authReducer } from '../auth/authReducer'

describe('<AppRouter /> test', () => {
	test('should show login if isnt authenticated', () => {
		const contextValue = {
			dispatch: jest.fn(),
			user: { logged: false },
		}
		const wrapper = mount(
			<AuthContext.Provider value={contextValue}>
				<AppRouter />
			</AuthContext.Provider>,
		)
		// console.log(wrapper.html())
		expect(wrapper).toMatchSnapshot()
	})

	test('should show <MarvelScreen /> if is authenticated', () => {
		const contextValue = {
			dispatch: jest.fn(),
			user: { logged: true, name: 'Juan' },
		}
		const wrapper = mount(
			<AuthContext.Provider value={contextValue}>
				<AppRouter />
			</AuthContext.Provider>,
		)
		// console.log(wrapper.html())
		// expect(wrapper).toMatchSnapshot()
		expect(wrapper.find('.navbar').exists()).toBe(true)
	})
})
