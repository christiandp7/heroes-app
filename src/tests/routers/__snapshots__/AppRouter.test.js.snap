// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> test should show login if isnt authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
  </div>
</div>
`;
